Implement vue math

AustinMroz opened the pull request:
Adds support for entering math inside number widgets in vue mode
![vue-math_00001](https://github.com/user-attachments/assets/e8ed7e2a-511f-4550-bfdd-1c467972d30d)

Migrates components to simple html elements (div and button) by borrowing styling from the (reverted) reka-ui migration in #6985. The existing (evil) litegraph eval code is extracted as a utility function and reused.

This PR means we're entirely writing our own NumberField.

Also adds support for scrubbing widgets like in litegraph
![scrubbing_00001](https://github.com/user-attachments/assets/890bcd28-34f4-4be0-908f-657e43f671b0)

### Known Issue
- ~~Scrubbing causes text to be highlighted~~, ~~starting a scrub from highlighted text will instead drag the text~~.
  - Resolved with the help of Alex

(Obligatory: _It won't do you a bit of good to review math_)

┆Issue is synchronized with this [Notion page](https://www.notion.so/PR-7759-Implement-vue-math-2d46d73d365081b9acd4d6422669016e) by [Unito](https://www.unito.io)

AustinMroz committed:
Implement vue math

qudix commented:
<img width="184" height="146" alt="Screenshot 2025-12-24 210635" src="https://github.com/user-attachments/assets/760edc06-c144-4d28-959b-42e460322277" />

AustinMroz committed:
fix tests

AustinMroz committed:
Copy props that would be provided by reka ui

aria-label="Decrease", aria-roledescription="Number field", and
aria-label="Increase" were skipped as linting requires localization, but
these do not appear to have been before. I lean towards the caution of
"no aria is better than wrong aria" here.

AustinMroz committed:
Add value scrubbing

This removes the ability to highlight the text value

AustinMroz committed:
No prevent mousedown, nits

Preventing mouseDown prevents focus at location of mouse click. This
feels awful and has been reverted.

Minor naming tweaks

Remove unused style

AustinMroz committed:
Handle pointer cancel

AustinMroz committed:
Fix disable on link connected

AustinMroz committed:
Fix test, allow key events to open control popover

AustinMroz committed:
nits

AustinMroz committed:
Fix international support, tighten regex

AustinMroz committed:
Add period to regex group

Commit:
[automated] Update test expectations

AustinMroz committed:
Empty commit for tests

DrJKL committed:
Numeric widget input experiment: Scrub without select

DrJKL committed:
Different cursor for scrubbable

AustinMroz committed:
nits

DrJKL commented on src/lib/litegraph/src/utils/widget.ts:
```diff
@@ -8,3 +8,18 @@ import type { IWidgetOptions } from '@/lib/litegraph/src/types/widgets'
+  if (/^[\d\s.()*+/-]+$/.test(input)) {
+    // Solve the equation if possible
+    try {
+      input = eval(input)
```
@coderabbitai Open an issue to replace the evals with a library, possibly Math.js

AustinMroz committed:
Check if value is finite

DrJKL commented on src/renderer/extensions/vueNodes/widgets/components/WidgetInputNumberInput.vue:
```diff
@@ -15,9 +16,58 @@ import WidgetLayoutField from './layout/WidgetLayoutField.vue'
 }>()
+const { locale } = useI18n()
+
+const decimalSeparator = computed(() =>
```
I do have a preference. It ended up being wrong. You've already fixed it 😁

DrJKL commented on src/renderer/extensions/vueNodes/widgets/components/WidgetInputNumberInput.vue:
```diff
@@ -69,6 +127,48 @@ const buttonsDisabled = computed(() => {
+    textEdit.value = true
+    inputField.value?.focus()
+    inputField.value?.setSelectionRange(0, -1)
+  }
```
https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/setSelectionRange

Commit:
[automated] Update test expectations

DrJKL commented on src/lib/litegraph/src/utils/widget.ts:
```diff
@@ -8,3 +8,18 @@ import type { IWidgetOptions } from '@/lib/litegraph/src/types/widgets'
+  const newValue = Number(input)
+  if (isNaN(newValue)) return undefined
+  return newValue
+}
```
See the issue opened related to this.

AustinMroz committed:
Empty commit for tests
